backend/src/filehandler/SSHFileHandler.ts

```diff
@@ -51,18 +52,30 @@ export default class SSHFileHandler {
     return new Promise((resolve, reject) => {
       this.client.sftp((err, sftp) => {
         if (err) return reject(err);
-        console.log("Retrieved sftp.");
+        console.log("Deployed via sftp.");
         const writeStream = sftp.createWriteStream(absolutePath);
         writeStream.write(content);
         writeStream.on("close", () => {
           console.log("stream closed 1");
+          sftp.end();
           resolve();
         });
         writeStream.on("finish", () => {
+          sftp.end();
           console.log("stream closed 2");
         });
         writeStream.end();
       });
     });
   }
+
+  async close(): Promise<void> {
+    if (this.hasClosedOrErrored) {
+      return Promise.reject(new Error("SSHFileHandler: SSH connection error."));
+    }
+    return new Promise((resolve) => {
+      this.client.end();
```

Comment:
There is no async operation here that requires this promise to be returned. You can just use this.client.end and return nothing.

Reply:
Thanks! Thought a Promise should to be returned, but make sense! Fixed